staging/src/k8s.io/apiserver/pkg/storageversion/updater.go

```diff
@@ -122,33 +123,21 @@ func setStatusCondition(conditions *[]v1alpha1.StorageVersionCondition, newCondi
 	existingCondition.ObservedGeneration = newCondition.ObservedGeneration
 }
 
-// updateStorageVersionFor updates the storage version object for the resource.
-func updateStorageVersionFor(c Client, apiserverID string, gr schema.GroupResource, encodingVersion string, decodableVersions []string, servedVersions []string) error {
-	retries := 3
-	var retry int
-	var err error
-	for retry < retries {
-		err = singleUpdate(c, apiserverID, gr, encodingVersion, decodableVersions, servedVersions)
-		if err == nil {
-			return nil
-		}
-		if apierrors.IsAlreadyExists(err) || apierrors.IsConflict(err) {
-			time.Sleep(1 * time.Second)
-			continue
-		}
-		if err != nil {
-			klog.Errorf("retry %d, failed to update storage version for %v: %v", retry, gr, err)
-			retry++
-			time.Sleep(1 * time.Second)
-		}
+// UpdateStorageVersionFor updates the storage version object for the resource.
+func UpdateStorageVersionFor(ctx context.Context, c Client, apiserverID string, gr schema.GroupResource, encodingVersion string, decodableVersions []string, servedVersions []string, postProcessFunc processStorageVersionFunc) error {
+	err := singleUpdate(ctx, c, apiserverID, gr, encodingVersion, decodableVersions, servedVersions, postProcessFunc)
+	if err != nil {
+		time.Sleep(1 * time.Second)
```

Comment:
Ah no, this was leftover from when I moved the retry logic from here to [here](https://github.com/kubernetes/kubernetes/pull/120582/files/7929a9aa97f42c0872a3c5f37979e76a06128241#diff-01e7a2059fe62e95ec32e2589c01b7218100de327647c641c334d38f2c5aa06eR1175-R1192)

But now that I look at it, it seems that retry block didn't need to be moved really, since UpdateStorageVersionFor() already does perform the retries. Reverted the retry block move.

Reply:
After reading your comment I expected the sleep to be removed/reverted; but its still here. Is that intentional

Reply:
I reverted back the retry block so that it remains in-place, and so does the time.sleep. (or am I not seeing the right changes on my end?)
The only change that's done to UpdateStorageVersionFor() is add ctx and processStorageVersionFunc params now.